modules/gmc-agent/src/main/java/org/gridgain/service/tracing/TracingService.java

```diff
@@ -34,6 +33,12 @@
  * Tracing service.
  */
 public class TracingService implements AutoCloseable {
+    /** Queue capacity. */
+    private static final int QUEUE_CAP = 100;
+
+    /** Logger. */
+    private IgniteLogger log;
```

Comment:
Can `log`, `ctx` and `onNodeTraces` be final?

Reply:
Done.